### PR TITLE
Most options

### PR DESCRIPTION
**Pull request by FG-TUM**
# Description

Sometimes we do not want all options but only those that serve as sane defaults. Therefore `getMostOptions` is added to Options.

**WARNING** This PR changes the default behavior of the `AutoPas` class from `getAllOptions()` to `getMostOptions`! See 4ce04a88.

## Related Pull Requests

- [ls1 #127](https://github.com/ls1mardyn/ls1-mardyn/pull/127)
- I would like to wait after #469 is merged to also use `getMostOptions()` as defaults for the MDFlexConfig.

## ~Resolved Issues~

# ~How Has This Been Tested?~

### CODE REVIEW

**Comment by FG-TUM** [comment]
@SteffenSeckler I would actually like this function to be in the `Option.h` interface. However, since this should be a static function and every option needs to have individual behavior I see no good way to do this. The options that came to my mind:
1. static virtual function -> not possible
2. static function with `if(std::is_same_v<actualOption, ContainerOption>) {...}` -> produces circular dependencies.

Any ideas?

File: examples/md-flexible/parsing/MDFlexConfig.h [hunk 1]
```diff
@@ -61,7 +61,7 @@ class MDFlexConfig {
 
   // AutoPas options:
   static inline const char *containerOptionsStr{"container"};
-  std::set<autopas::ContainerOption> containerOptions{autopas::ContainerOption::getAllOptions()};
+  std::set<autopas::ContainerOption> containerOptions{autopas::ContainerOption::getMostOptions)};
```
Comment (SteffenSeckler) [inline examples/md-flexible/parsing/MDFlexConfig.h:64]:
```suggestion
  std::set<autopas::ContainerOption> containerOptions{autopas::ContainerOption::getMostOptions()};
```

**Comment by SteffenSeckler** [comment]
static "virtual" is actually possible, as you are using CRTP.

**Comment by SteffenSeckler** [comment]
does the change of default values have any effect on our tests?

**Comment by SteffenSeckler** [comment]
as a quick way to switch between default parameters for AutoPas, one could also create new function(s) `enableAllOptions()` and `enableMeaningfulOptions()` (or similarly named, potentially also only one function, then with parameter).

**Comment by FG-TUM** [comment]
So as in the description above mentioned this changes the default behavior. At the moment the only change is that the direct sum container is dropped from the defaults but this may change at any time.
**The important question is: Does this affect the tests?**
As far as I can tell no. All tests that rely on iterating over all options use `getAllOptions()` at some point and not the `AutoPas` object.
@SteffenSeckler please do a quick check if you find any test that behaves differently.

**Comment by SteffenSeckler** [comment]
somehow, you have also stumbled across a (not-so-new) data race:
in Thermostat.h of md-flexible, a default_random_engine is used by multiple threads concurrently.
According to:
https://stackoverflow.com/questions/42157381/thread-safety-of-stdrandom-device
this is not a good idea.

File: examples/md-flexible/parsing/MDFlexConfig.h [hunk 2]
```diff
@@ -61,7 +61,7 @@ class MDFlexConfig {
 
   // AutoPas options:
   static inline const char *containerOptionsStr{"container"};
-  std::set<autopas::ContainerOption> containerOptions{autopas::ContainerOption::getAllOptions()};
+  std::set<autopas::ContainerOption> containerOptions{autopas::ContainerOption::getMostOptions()};
   static inline const char *dataLayoutOptionsStr{"data-layout"};
   std::set<autopas::DataLayoutOption> dataLayoutOptions{autopas::DataLayoutOption::getAllOptions()};
```
[the remark this answers is not in the document]
Comment (FG-TUM) [reply examples/md-flexible/parsing/MDFlexConfig.h:66]: yes but as stated above i only want to tough those files after #469 is merged

**Comment by FG-TUM** [comment]
> somehow, you have also stumbled across a (not-so-new) data race:
> in Thermostat.h of md-flexible, a default_random_engine is used by multiple threads concurrently.
> According to:
> https://stackoverflow.com/questions/42157381/thread-safety-of-stdrandom-device
> this is not a good idea.
> 
> I think that newly seeing this now is because the DirectSum container is no longer used for the ThermostatTest.
> 
> Two possible ways out:
> a) remove the `pragma omp parallel` from `Thermostat::addBrownianMotion`
> b) add one random device per thread (see also the answer of the above stackoverflow thread on how to do this.)

True. Will be fixed in separate PR.

**Pull request merged**